selfservice/strategy/oidc/provider_google.go

```diff
@@ -66,3 +70,27 @@ func (g *ProviderGoogle) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
 
 	return options
 }
+
+var _ IDTokenVerifier = new(ProviderGoogle)
+
+func (p *ProviderGoogle) Verify(ctx context.Context, rawIDToken string) (*Claims, error) {
+	keySet := oidc.NewRemoteKeySet(ctx, p.JWKSUrl)
+	verifier := oidc.NewVerifier("https://accounts.google.com", keySet, &oidc.Config{
+		ClientID: p.config.ClientID,
+	})
+	token, err := verifier.Verify(oidc.ClientContext(ctx, otelhttp.DefaultClient), rawIDToken)
+	if err != nil {
+		return nil, err
+	}
+	claims := &Claims{}
+	if err := token.Claims(claims); err != nil {
+		return nil, err
+	}
+	return claims, nil
+}
+
+var _ NonceValidationSkipper = new(ProviderGoogle)
+
+func (a *ProviderGoogle) CanSkipNonce(c *Claims) bool {
+	return true // TODO!!
```

Comment:
This will cause Ory to only validate `nonce`s for Google if the ID Token from Google contains a nonce. 

Since the SDKs I saw, don't support setting `nonce`s, this is the best solution I can come up, without excluding a bunch of SDKs here.

Reply:
Ok this is fine for me. Please document this as a caveat though when writing docs for this!

Reply:
I adjusted it to only return true, if no nonce is present in the claims.